wkcuber/converter.py

```diff
@@ -0,0 +1,283 @@
+        put_default_if_not_present(args, "distribution_strategy", "multiprocessing")
+        put_default_if_not_present(args, "job_resources", None)
+        put_default_if_not_present(args, "pad", False)
+        put_default_if_not_present(args, "max_mag", 32)
+        put_default_if_not_present(args, "no_compress", False)
+        put_default_if_not_present(args, "isotropic", False)
+        put_default_if_not_present(args, "verbose", True)
+
+        # detect layer and ds name
+        dataset_name, layer_names = self.detect_dataset_and_layer_names()
+        put_default_if_not_present(args, "name", dataset_name)
+
+        self.args = args
+
+        # TODO we can detect the layer names correctly, but we currently do not split the source files accordingly
+        args.layer_name = layer_names[0]
+        convert_image_stack(args)
+
+    def detect_dataset_and_layer_names(self) -> Tuple[str, List[str]]:
+        paths = set()
+        for f in self.source_files:
+            p = path.dirname(f)
+            if p != "":
+                paths.add(p)
+
+        if len(paths) == 0:
+            # if no parent folder is found, it is implicitly clear that there is only one source file
+            return path.splitext(path.basename(self.source_files[0]))[0], ["color"]
+
+        one_path = paths.pop()
+        dataset_name = path.dirname(one_path)
+        assert all(
+            map(lambda p: path.dirname(p) == dataset_name, paths)
+        ), "Not all image files can be ordered into layers"
+        if dataset_name == "":
+            # this means that all source files are from one folder
+            dataset_name = path.basename(one_path)
+            if dataset_name in ["color", "segmentation", "mask"]:
+                return path.basename(self.args.target_path), [dataset_name]
+            elif len(self.source_files) == 1:
+                return dataset_name, [
+                    path.splitext(path.basename(self.source_files[0]))[0]
+                ]
+            else:
+                return dataset_name, ["color"]
+        else:
+            dataset_name = path.basename(dataset_name)
+            paths.add(one_path)
+            layer_names = list(map(lambda p: cast(str, path.basename(p)), paths))
+            return dataset_name, layer_names
+
+
+class ConverterManager:
+    def __init__(self) -> None:
+        self.converter: List[Converter] = [
+            WkwConverter(),
+            NiftiConverter(),
+            KnossosConverter(),
+            ImageStackConverter(),
+        ]
+
+
+converter_manager = ConverterManager()
+
+if __name__ == "__main__":
+    args = create_parser().parse_args()
+
+    fitting_converters = list(
+        filter(
+            lambda c: c.accepts_input(args.source_path),
+            converter_manager.converter,
+        )
+    )
+
+    if len(fitting_converters) == 0:
+        print(
+            "No converter found. Please specify which converter you want to use or check the source path."
```

Comment:
How would one specify the converter?

Reply:
This was an initial idea of me that you could overwrite the converter checking. However, this should probably be removed or do you think there is a use case for that?

Reply:
I think, there is no strong need for this, since we also have the other high-level entry points for conversion. In the long run, it might make sense to unify all of this, so there is simply `python -m cuber convert ...` with optional `--source_type tiff`. But that's not important for now imo.